scrapy/utils/reqser.py

```diff
@@ -20,6 +19,7 @@ def request_to_dict(request, spider=None):
     if callable(eb):
         eb = _find_method(spider, eb)
     d = {
+        '__class__': request.__class__,
```

Comment:
That was fast! I made a few modifications to this PR and now it stores a string pointing to the class, this allows the dict to be serialized using `Marshal`-based queues. It also works well with requests without `__class__` key, falling back to the default `Request`, but it is true that we would save some resources saving the class only when it's not `Request`; let me patch that.